cypress/support/commands.js

```
@@ -1,70 +1,49 @@
-// ***********************************************
-// This example commands.js shows you how to
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-
 const imgUrl = 'https://static.productionready.io/images/smiley-cyrus.jpg';
 
-Cypress.Commands.add('login', (email, username, password) => {
-  cy.request('POST', '/api/users', {
-    user: {
-      email,
-      username,
-      password
-    }
-  }).then((response) => {
-    const user = {
-      bio: response.body.user.bio,
-      effectiveImage: imgUrl,
-      email: response.body.user.email,
-      image: response.body.user.image,
-      token: response.body.user.token,
-      username: response.body.user.username
-    };
-    window.localStorage.setItem('user', JSON.stringify(user));
-    cy.setCookie('auth', response.body.user.token);
+// Register a user (only once)
+Cypress.Commands.add('register', (email, username, password) => {
+  return cy.request({
+    method: 'POST',
+    url: '/api/users',
+    body: { user: { email, username, password } },
+    failOnStatusCode: false, // optional: ignore if user exists
```

`failOnStatusCode: false` will prevent Cypress from failing the command when the registration response is an error. That can hide real failures (not only "user exists"). Since you create unique emails this is generally safe, but I recommend either removing the `failOnStatusCode: false` or explicitly checking the response status after `cy.register` to ensure expected behavior and fail fast on unexpected errors.